utils/check_code_style.py

```diff
@@ -71,7 +71,9 @@
 	# Matches any tabulator characters.
 	"\t": "tabulators should only be used for indentation",
 	# Matches any commas that are not followed by whitespace characters.
-	",\\S": "commas should be followed by whitespaces"
+	",\\S": "commas should be followed by whitespaces",
+	# Matches incorrect structured bindings
+	r"\s&\s+\[": "structured bindings should not contain a space after the ampersand: &[first, second]",
```

Comment:
This is how the other regex strings are written in here.
```suggestion
	"\\s&\\s+\\[": "structured bindings should not contain a space after the ampersand: &[first, second]",
```
Also, this will only match a sequence like `& [` is it is preceded by whitespace, such as in `auto & [`, but we should also match cases where the ampersand is not separated from the type: `auto& [`, where it is not preceded by whitespace.
```suggestion
	"&\\s+\\[": "structured bindings should not contain a space after the ampersand: &[first, second]",
```